src/ORM/Entity.php

```diff
@@ -103,6 +105,45 @@ public function fill(array $data): void
         }
     }
 
+    /**
+     * Generate a unique value based on provided field.
+     *
+     * @param \EoneoPay\Externals\ORM\EntityManager $entityManager
+     * @param string $field
+     * @param int|null $length
+     *
+     * @return string
+     *
+     * @throws \EoneoPay\Externals\ORM\Exceptions\ORMException
+     * @throws \EoneoPay\Externals\ORM\Exceptions\RepositoryClassNotFoundException
+     */
+    public function generateRandomUniqueValue(
```

Comment:
This feature should be on the `EntityManagerInterface`, you give it the entity class, field, and length the same way. But we need to keep the entity as dumb as possible.